File: system-test/typescript/src/testkit/dsl/port/myShop/when/steps/when-place-order.ts
import type { ThenResultStage } from '../../then/then-result-stage.js';

export interface WhenPlaceOrder {
  withOrderNumber(orderNumber: string): this;
  withSku(sku: string): this;
  withQuantity(quantity: string | number): this;
  withCountry(country: string): this;
  withCouponCode(couponCode?: string | null): this;
  then(): ThenResultStage;
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/port/when/steps/WhenPlaceOrder.java
package com.mycompany.myshop.testkit.dsl.port.when.steps;

import com.mycompany.myshop.testkit.dsl.port.when.steps.base.WhenStep;

public interface WhenPlaceOrder extends WhenStep {
    WhenPlaceOrder withOrderNumber(String orderNumber);

    WhenPlaceOrder withSku(String sku);

    WhenPlaceOrder withQuantity(String quantity);

    WhenPlaceOrder withQuantity(int quantity);

    WhenPlaceOrder withCountry(String country);

    WhenPlaceOrder withCouponCode(String couponCode);

    WhenPlaceOrder withCouponCode();
}
